generative_model_training/05_plot_labels_sampling_performance.py

```diff
@@ -0,0 +1,511 @@
+        color="label_source",
+    )
+    + p9.geom_point()
+    + p9.geom_line()
+    + p9.geom_errorbar()
+    + p9.facet_wrap("~ prediction_label")
+    + p9.theme_bw()
+    + p9.theme(figure_size=(10, 10))
+    + p9.scale_color_brewer(type="qual", palette=2)
+)
+print(g)
+
+# ## Area Under Receiver Operating Curve (AUROC)
+
+# ### Tune Set
+
+g = (
+    p9.ggplot(
+        performance_ci_df
+        >> ply.query("data_source=='abstract'")
+        >> ply.query("model=='tune'")
+    )
+    + p9.aes(
+        x="lf_num",
+        y="auroc_mean",
+        ymin="auroc_lower_ci",
+        ymax="auroc_upper_ci",
+        group="label_source",
+        color="label_source",
+    )
+    + p9.geom_point()
+    + p9.geom_line()
+    + p9.geom_errorbar()
+    + p9.facet_wrap("~ prediction_label")
+    + p9.theme_bw()
+    + p9.theme(figure_size=(10, 10))
+    + p9.scale_color_brewer(type="qual", palette=2)
+)
+print(g)
+
+# ### Test Set
+
+g = (
+    p9.ggplot(
+        performance_ci_df
+        >> ply.query("data_source=='abstract'")
+        >> ply.query("model=='test'")
+    )
+    + p9.aes(
+        x="lf_num",
+        y="auroc_mean",
+        ymin="auroc_lower_ci",
+        ymax="auroc_upper_ci",
+        group="label_source",
+        color="label_source",
+    )
+    + p9.geom_point()
+    + p9.geom_line()
+    + p9.geom_errorbar()
+    + p9.facet_wrap("~ prediction_label")
+    + p9.theme_bw()
+    + p9.theme(figure_size=(10, 10))
+    + p9.scale_color_brewer(type="qual", palette=2)
+)
+print(g)
+
+# ## Binary Cross Entropy Loss
+
+# ### Tune Set
+
+g = (
+    p9.ggplot(
+        performance_ci_df
+        >> ply.query("data_source=='abstract'")
+        >> ply.query("model=='tune'")
+    )
+    + p9.aes(
+        x="lf_num",
+        y="bce_mean",
+        ymin="bce_lower_ci",
+        ymax="bce_upper_ci",
+        group="label_source",
+        color="label_source",
+    )
+    + p9.geom_point()
+    + p9.geom_line()
+    + p9.geom_errorbar()
+    + p9.facet_wrap("~ prediction_label")
+    + p9.theme_bw()
+    + p9.theme(figure_size=(10, 10))
+    + p9.scale_color_brewer(type="qual", palette=2)
+)
+print(g)
+
+# ### Test Set
+
+g = (
+    p9.ggplot(
+        performance_ci_df
+        >> ply.query("data_source=='abstract'")
+        >> ply.query("model=='test'")
+    )
+    + p9.aes(
+        x="lf_num",
+        y="bce_mean",
+        ymin="bce_lower_ci",
+        ymax="bce_upper_ci",
+        group="label_source",
+        color="label_source",
+    )
+    + p9.geom_point()
+    + p9.geom_line()
+    + p9.geom_errorbar()
+    + p9.facet_wrap("~ prediction_label")
+    + p9.theme_bw()
+    + p9.theme(figure_size=(10, 10))
+    + p9.scale_color_brewer(type="qual", palette=2)
+)
+print(g)
+
+# # Full Text
+
+# ## Area Under Precision Recall Curve (AUPR)
+
+# ### Tune Set
+
+g = (
+    p9.ggplot(
+        performance_ci_df
+        >> ply.query("data_source=='full_text'")
+        >> ply.query("model=='tune'")
+    )
```

Comment:
Is there a reason that adding more CbG label functions to predict CbG makes the performance worse? Is it that there are only a few useful functions for that edge, so adding more functions increases noise?

Reply:
>Is there a reason that adding more CbG label functions to predict CbG makes the performance worse? Is it that there are only a few useful functions for that edge, so adding more functions increases noise?

Yes, you hit the nail on the head. The underlying problem here is it is HARD to tell if the label functions themselves need to be better calibrated or I just need to add more label functions to boost performance. My hunch is the first one as I wouldn't be surprised if majority of sentences are only being labeled by my distant supervision functions other than text pattern functions.